packages/core/saved-objects/core-saved-objects-base-server-internal/src/model_version/build_transform_fn.ts

```diff
@@ -20,12 +24,46 @@ export const buildModelVersionTransformFn = (
   const transformFns: SavedObjectModelTransformationFn[] = [];
   modelChanges.forEach((change) => {
     if (change.type === 'data_backfill') {
-      transformFns.push(change.transform);
+      transformFns.push(dataBackfillChangeToTransformFn(change));
+    }
+    if (change.type === 'data_removal') {
+      transformFns.push(dataRemovalChangeToTransformFn(change));
+    }
+    if (change.type === 'unsafe_transform') {
+      transformFns.push(unsafeTransformChangeToTransformFn(change));
```

Comment:
Converting the change types into actual `SavedObjectModelTransformationFn` functions